django/db/models/expressions.py

```diff
@@ -907,6 +928,63 @@ def convert_value(self, value, expression, connection, context):
         return value
 
 
+class SubQuery(Expression):
+    """
+    An explicit sub-query. It may contain OuterRef() references to the outer query,
+    which will be resolved when it is applied to that query.
+    """
+    template = '(%(subquery)s)'
+
+    def __init__(self, subquery, output_field=None, **extra):
+        self.subquery = subquery.all()
+        self.extra = extra
+        super(SubQuery, self).__init__(output_field)
+
+    def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize=False, for_save=False):
+        clone = self.copy()
+        clone.is_summary = summarize
+        # We need a new copy of the subquery too, because we will be modifying it.
+        clone.subquery = clone.subquery.all()
+        clone.subquery.query.bump_prefix(query)
+
+        # Need to recursively resolve these.
+        def resolve(child):
+            if hasattr(child, 'children'):
+                [resolve(_child) for _child in child.children]
+            if hasattr(child, 'rhs') and isinstance(child.rhs, F):
+                child.rhs = child.rhs.resolve_expression(query, allow_joins, reuse, summarize, for_save)
+
+        resolve(clone.subquery.query.where)
+        return clone
+
+    def get_source_expressions(self):
+        return [x for x in [getattr(expr, 'lhs', None) for expr in self.subquery.query.where.children] if x]
+
+    def as_sql(self, compiler, connection, template=None, **extra_context):
+        connection.ops.check_expression_support(self)
+        template_params = self.extra.copy()
+        template_params.update(extra_context)
+
+        template_params['subquery'], sql_params = self.subquery.query.sql_with_params()
```

Comment:
I had to change this to
`template_params['subquery'], sql_params = self.subquery.query.get_compiler(connection=connection).as_sql()`
in order to compile correctly when using a database other than the `DEFAULT_DB_ALIAS`.